src/shapes/textbox.class.ts

```diff
@@ -293,6 +304,83 @@ export class Textbox extends IText {
     return value.split(this._wordJoiners);
   }
 
+  /**
+   * Wraps a line of text using the width of the Textbox and a context. For overflowBreakWord option
+   * @param {Array} line The grapheme array that represent the line
+   * @param {Number} lineIndex
+   * @param {Number} desiredWidth width you want to wrap the line to
+   * @param {Number} reservedSpace space to remove from wrapping for custom functionalities
+   * @returns {Array} Array of line(s) into which the given text is wrapped
+   * to.
+   */
+  _wrapLineOfWordBreak(
+    _line,
+    lineIndex: number,
+    desiredWidth: number,
+    reservedSpace = 0
+  ): Array<any> {
+    desiredWidth -= reservedSpace;
+    const graphemeLines = [];
+    let text = _line || '',
+      i = 0,
+      j = 0,
+      offset = 0,
+      temp = '',
+      width = 0,
+      prevGrapheme = '',
+      largestLetterWidth = 0;
+
+    if (!text.length) {
+      graphemeLines.push([]);
+    }
+    while (text.length > 0) {
+      i = 0;
+      // Get the maximum string at a fixed width
+      width = 0;
+      prevGrapheme = '';
+      for (let k = 0, len = text.length; k < len; k++) {
+        const box = this._getGraphemeBox(
+          text[k],
+          lineIndex,
+          offset + k,
+          prevGrapheme,
+          true
+        );
+        width += box.kernedWidth;
+        largestLetterWidth = Math.max(largestLetterWidth, box.kernedWidth);
+        if (width > Math.max(largestLetterWidth, desiredWidth)) {
+          i = k;
```

Comment:
Doesn't this make the last letter spill over the bbox?
Should it be `i = Math.max(k - 1, 0)` ?

Reply:
correct